cherrypy/lib/auth_digest.py

```diff
@@ -38,11 +43,43 @@ def md5_hex(s):
     return md5(ntob(s, 'utf-8')).hexdigest()
 
 
+def sha256_hex(s):
+    """Return hexdigest of sha256sum."""
+    return sha256(ntob(s, 'utf-8')).hexdigest()
+
+
+def sha512_hex(s):
+    """Return hexdigest of sha512sum."""
+    return sha512(ntob(s, 'utf-8')).hexdigest()
+
+
+def choose_hexdigest_method(algorithm):
+    """Return Python hexdigest method for given algorithm name."""
+    if algorithm.endwith('-sess'):
+        algorithm = algorithm.rsplit('-', 1)[0]
+    algorithm = algorithm.replace('-', '')
+    return algorithm.lower() + '_hex'
+
+
 qop_auth = 'auth'
 qop_auth_int = 'auth-int'
 valid_qops = (qop_auth, qop_auth_int)
 
-valid_algorithms = ('MD5', 'MD5-sess')
+
+valid_algorithms = {
+    'MD5',
+    'MD5-sess',
+    'SHA-256',
+    'SHA-256-sess',
+    'SHA-512',
+    'SHA-512-sess',
+}
+
+
+def updte_valid_algorithms(algorithms):
```

Comment:
There's a typo in the wrd updte here. Also, I don't think that it's acceptable to have a mechanism for manipulating global state like this.